api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -33,13 +34,15 @@ function _setSchoolingRegistrationFilters(qb, { lastName, firstName, connexionTy
   if (connexionType === 'none') {
     qb.whereRaw('"users"."username" IS NULL');
     qb.whereRaw('"users"."email" IS NULL');
-    qb.whereRaw('"users"."samlId" IS NULL');
+    // we only retrieve GAR authentication method in join clause
+    qb.whereRaw('"authentication-methods"."externalIdentifier" IS NULL');
   } else if (connexionType === 'identifiant') {
     qb.whereRaw('"users"."username" IS NOT NULL');
   } else if (connexionType === 'email') {
     qb.whereRaw('"users"."email" IS NOT NULL');
   } else if (connexionType === 'mediacentre') {
-    qb.whereRaw('"users"."samlId" IS NOT NULL');
+    // we only retrieve GAR authentication method in join clause
+    qb.whereRaw('"authentication-methods"."externalIdentifier" IS NOT NULL');
```

Comment:
Est-ce qu'il ne faudrait pas faire directement ici

```suggestion
    qb.whereRaw('"authentication-methods"."identityProvider" = AuthenticationMethod.identityProviders.GAR');
```

Pour éviter la confusion avec les autre identifiants externes (Pôle Emploi, ...)

Reply:
Ca complexifie la requête car je spécifie l'`identityProvider` GAR dans la jointure, donc je m'attends à avoir 0 ou 1 `authentication-mehod`. Le fait de les faire tous remonter a des impacts sur la génération du model pour simplifier uniquement la logique de filtre.

Reply:
Vu ensemble avec @MathieuGilet , puisqu'on est dans le repo des schooling-registration, on a jugé qu'on pouvait se permettre l'ambiguïté (assez faible) qui est levée juste après